### ISSUE TITLE
ha-experimental has two deployments

### ISSUE THREAD

**Issue by zfrhv** [kubernetes-sigs/hierarchical-namespaces#291]
when applying ha-experimental.yaml there are two deployments,
- one is single controller
- and other is ha controller with 3 replicas

total is 4 controller pods

is that on purpose or by accident?

**Comment by erikgb** [comment]
Yes I think so. At present the reconciler in HNC does not support leader election, so that is the single pod deployment. The deployment with 3 pods is to serve the webhooks - which is the most critical part of HNC (or any Kubernetes controller). But especially HNC, since one webhook is configured to match ALL namespaced resources.

**Comment by erikgb** [comment]
I am working on improving the "god" webhook configuration in https://github.com/kubernetes-sigs/hierarchical-namespaces/pull/285.

**Comment by zfrhv** [comment]
i see
thank you
